sec/audit: wait for 25.3 for audit rpc

pgellert opened the pull request:
The internal kafka RPCs are only available consistently after 25.3. Previously, they were only conditionally available, only if WASM is enabled, so we cannot rely on them being available during a 25.2 -> 25.3 upgrade.

Fixes https://redpandadata.atlassian.net/browse/CORE-14573

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [x] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

pgellert committed:
sec/audit: wait for 25.3 for audit rpc

The internal kafka RPCs are only available consistently after 25.3.
Previously, they were only conditionally available, only if WASM is
enabled, so we cannot rely on them being available during a 25.2 -> 25.3
upgrade.

pgellert commented on src/v/security/audit/audit_log_manager.cc:
```diff
@@ -123,11 +124,23 @@ audit_log_manager::audit_log_manager(
-    if (config::shard_local_cfg().audit_use_rpc()) {
+    const bool internal_rpcs_available
+      = _controller->get_feature_table().local().get_active_version()
+        >= features::to_cluster_version(features::release_version::v25_3_1);
```
I was considering "watching for" the upgrade to 25.3 and then immediately "restarting" the `audit_log_manager` and enabling the RPCs. That would be ideal, but that doesn't seem simple to implement. @oleiman @michael-redpanda do you think we would need that?

rockwotj replied:
I think this is OK

michael-redpanda reviewed (dismissed):
lgtm - thanks for addressing this

pgellert committed:
sec/audit: defer sink construction to start

This is needed for the next commit to ensure we can construct the sink
after the feature table has been initialized from the latest controller
snapshot.

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/89883f3e2d44ab3f8e81ddbc169a0252f0915bc6..374834e0056cb170dbf8d9e1d8c568dd3d868eb6): delay the active version check & sink construction to `start()` to do it after the feature table has been loaded from the controller snapshot

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--5c89dd7e-e89f-4ff8-80e8-e4f283dfe0d4-->
<!--TEST RESULTS-->
<!--3870bffa-0ef8-419b-b9c7-a37427b5118f-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75750">build#75750</a></summary>


| test_class            | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                             |
|:----------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_creating_and_listing_migrations | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75750#019a5988-1d91-42e8-8bfd-1af55a86fe73 | FLAKY         | 14/21    | upstream reliability is '98.77551020408163'. current run reliability is '66.66666666666666'. drift is 32.10884 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_creating_and_listing_migrations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75818">build#75818</a></summary>


| test_class                    | test_method                                  | test_arguments                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                  |
|:------------------------------|:---------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ReplicatedMetastoreTest       | TestBasicRemoveTopics                        |                                                                                                                                                | unit        | https://buildkite.com/redpanda/redpanda/builds/75818#019a5e48-4766-4a06-a354-ba527f501a33 | FAIL          | 0/1      |                                                                                                                                                                        |                                                                                                                                                                               |
| ShadowLinkingReplicationTests | test_replication_timestamps_match            | {"source_cluster_spec": {"cluster_type": "kafka", "kafka_quorum": "COMBINED_KRAFT", "kafka_version": "3.8.0"}, "timestamp_type": "CreateTime"} | integration | https://buildkite.com/redpanda/redpanda/builds/75818#019a5e6c-38de-40d1-be0f-82a74116c3e5 | FLAKY         | 20/21    | upstream reliability is '96.22425629290618'. current run reliability is '95.23809523809523'. drift is 0.98616 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_timestamps_match |
| EnterpriseFeaturesTest        | test_enable_features                         | {"disable_trial": false, "feature": 5, "install_license": false}                                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/75818#019a5e6c-38e2-4c9d-a59a-d5975537aa16 | FLAKY         | 20/21    | upstream reliability is '99.7584541062802'. current run reliability is '95.23809523809523'. drift is 4.52036 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EnterpriseFeaturesTest&test_method=test_enable_features                     |
| SegmentMsTest                 | test_segment_rolling_with_retention_consumer | null                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75818#019a5e95-2e31-4ba3-84fd-9fb0765841e9 | FLAKY         | 18/21    | upstream reliability is '94.24083769633508'. current run reliability is '85.71428571428571'. drift is 8.52655 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SegmentMsTest&test_method=test_segment_rolling_with_retention_consumer      |
</details>
<!--/body-->

michael-redpanda reviewed (commented):
Would like to see an upgrade test with this please

pgellert committed:
dt/audit: test auditing works mid-upgrade

- test on each individual node independently
- during the upgrade + after a rolling restart after the upgrade

pgellert commented:
[pushed](https://github.com/redpanda-data/redpanda/pull/28396/commits/ec955487c2046a1c1fbe5636e80c55af7dff27ad): an upgrade test for audit logging
